integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/BlueprintTest.java

```diff
@@ -0,0 +1,179 @@
+        String blueprintName = getNameGenerator().getInvalidRandomNameForMock();
+        testContext.given(BlueprintEntity.class)
+                .withName(blueprintName)
+                .withAmbariBlueprint(VALID_BP)
+                .when(Blueprint.postV4(), key(blueprintName))
+                .expect(BadRequestException.class, exceptionConsumer(e -> {
+                    assertThat(ResponseUtil.getErrorMessage(e), containsString("must match \"^[^;\\/%]*$\""));
+                }).withKey(blueprintName))
+                .validate();
+    }
+
+    @Test(dataProvider = "testContext")
+    public void testCreateBlueprintWithInvalidJson(TestContext testContext) {
+        String blueprintName = getNameGenerator().getRandomNameForMock();
+        testContext.given(BlueprintEntity.class)
+                .withName(blueprintName)
+                .withAmbariBlueprint("apple-tree")
+                .when(Blueprint.postV4(), key(blueprintName))
+                .expect(BadRequestException.class, exceptionConsumer(e -> {
+                    assertThat(ResponseUtil.getErrorMessage(e), containsString("Failed to parse JSON"));
+                }).withKey(blueprintName))
+                .validate();
+    }
+
+    @Test(dataProvider = "testContext")
+    public void testListBlueprint(TestContext testContext) {
+        testContext.given(BlueprintEntity.class)
+                .when(Blueprint.listV4())
+                .then(BlueprintTest::checkDefaultBlueprintsIsListed)
+                .validate();
+    }
+
+    @Test(dataProvider = "testContext")
+    public void testGetSpecificBlueprint(TestContext testContext) {
+        String blueprintName = getNameGenerator().getRandomNameForMock();
+        testContext.given(BlueprintEntity.class)
+                .withName(blueprintName)
+                .withAmbariBlueprint(VALID_BP)
+                .when(Blueprint.postV4(), key(blueprintName))
+                .when(Blueprint.getV4(), key(blueprintName))
+                .then((tc, entity, cc) -> {
+                    assertEquals(blueprintName, entity.getName());
+                    return entity;
+                })
+                .validate();
+    }
+
+    @Test(dataProvider = "testContext")
+    public void testDeleteSpecificBlueprint(TestContext testContext) {
+        String blueprintName = getNameGenerator().getRandomNameForMock();
+        testContext.given(BlueprintEntity.class)
+                .withName(blueprintName)
+                .withAmbariBlueprint(VALID_BP)
+                .when(Blueprint.postV4(), key(blueprintName))
+                .when(Blueprint.deleteV4(), key(blueprintName))
+                .then((tc, entity, cc) -> {
+                    assertEquals(blueprintName, entity.getName());
+                    return entity;
+                })
+                .when(Blueprint.listV4())
+                .then(new AssertionV2<BlueprintEntity>() {
```

Comment:
could you please move the assertion to a new method to make the test itself more readable?